pkg/server/server_builder.go

```diff
@@ -209,7 +209,7 @@ func (b *KiamServerBuilder) Build() (*KiamServer, error) {
 		credentialsProvider: credentialsCache,
 		assumePolicy: Policies(
 			NewRequestingAnnotatedRolePolicy(b.podCache, arnResolver),
-			NewNamespacePermittedRoleNamePolicy(b.namespaceCache, arnResolver),
+			NewNamespacePermittedRoleNamePolicy(!b.config.DisableStrictNamespaceRegexp, b.namespaceCache, arnResolver),
```

Comment:
Generally don't like subtle things like this where you end up with a double-negative...
I felt like the code should deal with a `strict` bool that was by default true, but the flag should be very verbose to make sure you understand you are about to reduce safety.